### PR TITLE
NEW stripe element with more gateways

### PR DESCRIPTION
**Pull request by ptibogxiv**
cf #https://github.com/Dolibarr/dolibarr/issues/19360

more doc here
https://stripe.com/docs/payments/payment-element

actually functionnal but car be optimized with less fields and better design

### CODE REVIEW

File: htdocs/public/payment/newpayment.php [hunk 1]
```diff
@@ -2423,49 +2422,28 @@
...
 
-					stripe.handleCardPayment(
-					clientSecret, cardElement, {
+					stripe.confirmPayment({
+		  elements,confirmParams: {
+		  return_url: '<?php echo $urlok; ?>',
 						payment_method_data: {
 							billing_details: {
-								name: cardholderName.value
+								name: 'test'
```
Comment (eldy) [inline htdocs/public/payment/newpayment.php:2446]: This looks strange. Does this means we do not propagate the name of credit card owner anymore ?

Comment (ptibogxiv) [reply htdocs/public/payment/newpayment.php:2446]: @eldy nope! The New element  form don’t use it

**Comment by eldy** [comment]
I modified your PR so to enable your new feature you must set 
STRIPE_USE_INTENT_WITH_AUTOMATIC_CONFIRMATION to 2 instead of 1
So the old code when STRIPE_USE_INTENT_WITH_AUTOMATIC_CONFIRMATION = 1 is still available.
And I keep this old code as the default value because the new one seems to not work correctly. I am investigating on it...

**Comment by ptibogxiv** [comment]
llok an eye on https://github.com/Dolibarr/dolibarr/pull/19444 i fix incompatibility between gateways...

**Comment by eldy** [comment]
This is not a problem in Stripe nor in TakePOS, but a problem in the new code implementation:
Take the new code, set STRIPE_USE_INTENT_WITH_AUTOMATIC_CONFIRMATION to 2 to use the new Stripe system. Then try to make a payment of a simple invoice using the link suggested on the invoice:
With old method (STRIPE_USE_INTENT_WITH_AUTOMATIC_CONFIRMATION = 1), after a payment is done the page newpayment.php was called with parameter action=charge and this page was sets sessions variables and then make a redirect to ko or ko page.
With the new one (STRIPE_USE_INTENT_WITH_AUTOMATIC_CONFIRMATION = 2), once a payment is done, the page  ok is called directly without the object sessions required by the paymentok page (data are not set into session like it is into the action=charge when STRIPE_USE_INTENT_WITH_AUTOMATIC_CONFIRMATION=1), so the paymentok page does nothing instead of recording the payment and other actions. So payment is validated into Stripe but not into Dolibarr !

**Comment by eldy** [comment]
I pushed a fix. I let you test that everything is working as expected...

**Pull request merged**